src/hotspot/share/runtime/os.hpp

```diff
@@ -311,6 +311,7 @@ class os: AllStatic {
   }
 
   static julong available_memory();
+  static julong free_memory();
```

Comment:
It would probably be a good idea to add a comment describing what the actual difference is between free/available (i.e. Linux only and usually `available memory > free memory` on such systems).

Reply:
Yes some detailed commentary is very necessary. It looks very odd to have two functions that do the same thing on every platform but Linux.

Reply:
I added comments into os_linux.hpp . I will move them to os.hpp .
https://github.com/openjdk/jdk/pull/13398/files#diff-b6c4026228694834053813a8a8ea4795b5edb29eeb9cc46cd234c8e3b92336f4R61-R64

Reply:
This looks fine to me.